ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -703,8 +706,11 @@ getMempoolReader mempool = MempoolReader.TxSubmissionMempoolReader
                                       snapshotHasTx } =
       MempoolReader.MempoolSnapshot
         { mempoolTxIdsAfter = \idx ->
-            [ (txId (txForgetValidated tx), idx', getTxSize mempool (txForgetValidated tx))
-            | (tx, idx') <- snapshotTxsAfter idx
+            [ ( txId (txForgetValidated tx)
+              , idx'
+              , fromIntegral $ unByteSize byteSize   -- TODO overflow?
```

Comment:
a TODO here

Reply:
Bah. I just lost a really nice draft in this message. I don't have the heart to re-write it right now :'(

summary:

- Fraser raised a concern about encoding sizes in LocalTxMonitor replies
- Chatted with Esgen and distilled down a very simple concrete correctness argument
- `Natural` trivializes the correctness of all _other_ logic; it unfortunately merely defers the correctness concerns to the encoder (... which is "_merely_" for a Node-to-Client QoL feature, but still quite important)
- the mempool should (carefully!) check for overflow of the resulting mempool size (resulting in `NoSpaceLeft`) before adding a tx to the finger tree; this inductively ensures that the Word64s in the mempool have never overflowed
    - if the mempool capacity is specified as a Word64, then a (careful!) mempool capacity check would imply this
- keep the new `txMeasure` partiality; otherwise an invalid tx (eg of size `maxBound`, or even just ten times per-block limit) could force an indefinite `NoSpaceLeft`
- ExUnits are inherently "big", but even 100*1024 times the current block size limit is still less than 2^50.